gini/gini-capture-sdk-ios
# Feature/pia 1798 send errors to pay api

Sends errors including error details to the Gini Pay API.

Please note: this build requires the Gini Pay API Lib [error logging PR](https://github.com/gini/gini-pay-api-lib-ios/pull/15) to be released and until then the builds won't succeed.

### How to test
Until the Gini Pay API Lib [error logging PR](https://github.com/gini/gini-pay-api-lib-ios/pull/15) has not been released you will need to check out the `feature/PIA-1690-error-logging` branch of the Pay API Lib and then add it to the example's Podfile with a path pointing to your local Pay API Lib folder:
```
pod 'GiniPayApiLib/DocumentsAPI', :path => '../../gini-pay-api-lib-ios/'
```

An error can be induced by replacing this line in the `DocumentService`:
```
self.fetchExtractions(for: partialDocumentsInfoSorted, completion: completion)
```
with:
```
self.fetchExtractions(for: [], completion: completion)
```
This will cause the composite document creation to return a failure response.

To see the logged error log into [Graylog](http://log.prod.gini.net/search?q=&rangetype=relative&relative=300) and use this search query: `_exists_:captureSdkVersion`.

### Ticket 
https://ginis.atlassian.net/browse/PIA-1798
